src/uwtools/rocoto.py

```diff
@@ -187,9 +187,18 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
             tag, _ = self._tag_name(key)
             if tag == STR.taskdep:
                 self._set_attrs(SubElement(e, STR.taskdep), block)
+            elif tag == STR.datadep:
+                self._add_data_dependency(e, config[STR.datatdep], block)
             else:
                 raise UWConfigError("Unhandled dependency type %s" % tag)
 
+    def _add_data_dependency(self, e: Element, config: dict, block: dict) -> None:
```

Comment:
That makes perfect sense to me. I actually went through `test_rocoto` and found `_add_task_dependency` was quite deeply embedded in all the tests already and that out of simplicity it might just be easier to keep it the way it is currently and just factor out `taskdep` where appropriate.